server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1196,13 +1203,21 @@ public void run() {
     } catch (InterruptedException e) {
       log.debug("Interrupted while fetching status");
     }
+
+    tp.shutdownNow();
+
+    // Because result is a ConcurrentSkipListMap will not see a concurrent modification exception,
+    // even though background threads may still try to put.
+    SortedMap<TServerInstance,TabletServerStatus> info = ImmutableSortedMap.copyOf(result);
```

Comment:
If something isn't in this map by the time we get here, we could just add the remaining directly to the badServers.

Reply:
The bad server processing needs some attention, I will open up a follow on issue.